src/state_machine/coordinator/fire.rs

```diff
@@ -271,7 +273,21 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
                     return Ok((Some(packet), None));
                 }
                 State::DkgEndGather => {
-                    self.gather_dkg_end(packet)?;
+                    if let Err(error) = self.gather_dkg_end(packet) {
+                        match error {
```

Comment:
Just a nit. Since this is a match on a single type (often clippy complains about this too), would change to

```rust
if let Error::DkgFailure(dkg_failure) = error {
    return Ok((...))
} else {
    return Err(error);
}
```

Reply:
Done.  I'm kinda surprised `clippy` didn't complain.